docs/server-event-types.md

```diff
@@ -0,0 +1,49 @@
+---
+id: server-event-types
+title: Event type reference
+sidebar_label: Event types
+---
+
+Event types are defined in the Server API [event_type.proto](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/event_type.proto) file.
+
+- **EVENT_TYPE_UNSPECIFIED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_SCHEDULED**:
+- **EVENT_TYPE_WORKFLOW_TASK_STARTED**:
+- **EVENT_TYPE_WORKFLOW_TASK_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_FAILED**:
+- **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**:
+- **EVENT_TYPE_ACTIVITY_TASK_STARTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_COMPLETED**:
+- **EVENT_TYPE_ACTIVITY_TASK_FAILED**:
+- **EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCELED**:
+- **EVENT_TYPE_TIMER_STARTED**:
+- **EVENT_TYPE_TIMER_FIRED**:
+- **EVENT_TYPE_TIMER_CANCELED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
```

Comment:
- **EVENT_TYPE_WORKFLOW_EXECUTION_CANCEL_REQUESTED**:
This event type indicates the workflow should be cancelled.